packages/harmony/src/components/button/FollowButton/FollowButton.tsx

```diff
@@ -0,0 +1,122 @@
+import { useState, useCallback, ChangeEvent } from 'react'
+
+import { useTheme, type CSSObject } from '@emotion/react'
+
+import type { IconComponent } from 'components/icon'
+import { Flex } from 'components/layout/Flex'
+import { Text } from 'components/text/Text'
+import { useControlled } from 'hooks/useControlled'
+import { IconUserFollowing, IconUserFollow, IconUserUnfollow } from 'icons'
+
+import type { FollowButtonProps } from './types'
+
+const messages = {
+  follow: 'Follow',
+  following: 'Following',
+  unfollow: 'Unfollow'
+}
+
+/**
+ * Special button for following or unfollowing a user.
+ */
+export const FollowButton = (props: FollowButtonProps) => {
+  const {
+    variant = 'default',
+    following = false,
+    onUnfollow,
+    onFollow,
+    size = 'default',
+    ...inputProps
+  } = props
+  const [value, setValueState] = useControlled({
+    componentName: 'FollowButton',
+    controlledProp: following,
+    defaultValue: undefined,
```

Comment:
either we provide tdefaultFollowing from props or just dont include this?

Reply:
the type requires that it be there, but I don't think we need to expose that functionality for now. Maybe that means we shouldn't be using useControlled? Like do we ever want a follow button that isn't fully controlled?